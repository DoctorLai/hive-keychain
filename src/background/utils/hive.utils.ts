import RPCModule from '@background/rpc.module';
import {
  Asset,
  ClaimRewardBalanceOperation,
  PrivateKey,
} from '@hiveio/dhive/lib/index-browser';
import { ActiveAccount } from '@interfaces/active-account.interface';

const claimRewards = async (
  activeAccount: ActiveAccount,
  rewardHive: string | Asset,
  rewardHBD: string | Asset,
  rewardVests: string | Asset,
): Promise<boolean> => {
  try {
    const client = await RPCModule.getClient();
    await client.broadcast.sendOperations(
      [
        [
          'claim_reward_balance',
          {
            account: activeAccount.name,
            reward_hive: rewardHive,
            reward_hbd: rewardHBD,
            reward_vests: rewardVests,
          },
        ] as ClaimRewardBalanceOperation,
      ],
      PrivateKey.fromString(activeAccount.keys.posting as string),
    );

    return true;
  } catch (err: any) {
    return false;
  }
};

const BgdHiveUtils = {
  claimRewards,
};

export default BgdHiveUtils;
