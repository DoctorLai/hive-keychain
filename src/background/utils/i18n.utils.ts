const getMessage = async (name: string) => {
  const lang = chrome.i18n.getUILanguage().split('-')[0];
  let res;
  try {
    const url = chrome.runtime.getURL(`_locales/${lang}/messages.json`);
    const file = await (await fetch(url)).json();
    res = file[name].message;
  } catch (e) {
    const urlEn = chrome.runtime.getURL(`_locales/en/messages.json`);
    const file = await (await fetch(urlEn)).json();

    res = file[name] ? file[name].message : `[Missing ${name} locale]`;
  } finally {
    //Logger.log('[i18n]', lang, name, res);
    return res;
  }
};

export default getMessage;
