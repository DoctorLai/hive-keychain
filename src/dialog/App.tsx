import { DialogCommand } from '@reference-data/dialog-message-key.enum';
import React, { useEffect, useState } from 'react';
import Unlock from 'src/dialog/pages/unlock';
import './dialog.scss';

const App = () => {
  useEffect(() => {
    chrome.runtime.onMessage.addListener(function (data, sender, sendResp) {
      setData(data);
    });
  }, []);

  const [data, setData] = useState<any>({});

  const renderDialogContent = () => {
    console.log(data);
    switch (data.command) {
      case DialogCommand.UNLOCK:
      case DialogCommand.WRONG_MK:
        return (
          <Unlock
            data={data}
            wrongMk={data.command === DialogCommand.WRONG_MK}
          />
        );
      default:
        return null;
    }
  };

  return <div className="dialog">{renderDialogContent()}</div>;
};

export default App;
