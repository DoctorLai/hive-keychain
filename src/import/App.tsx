import React, { Component } from 'react';
import { BackgroundCommand } from 'src/reference-data/background-message-key.enum';
import FileUtils from 'src/utils/file.utils';
import './import-keys.scss';

interface ImportAccountState {
  selectedFile?: File;
}

class App extends Component {
  state: ImportAccountState = {};

  handleFileUpload = (event: any) => {
    this.state.selectedFile = event.target.files[0];
  };

  importKeysFromFile = async () => {
    if (this.state.selectedFile) {
      const base64 = await FileUtils.toBase64(this.state.selectedFile);
      const fileData = atob(base64);
      chrome.runtime.sendMessage({
        command: BackgroundCommand.IMPORT_ACCOUNTS,
        value: fileData,
      });
    }
  };

  render() {
    return (
      <div className="import-keys">
        <div className="title-panel">
          <img src="/assets/images/iconhive.png" />
          <div className="title">
            {chrome.i18n.getMessage('import_html_title')}
          </div>
        </div>
        <div
          className="caption"
          dangerouslySetInnerHTML={{
            __html: chrome.i18n.getMessage('import_html_text'),
          }}></div>

        <input
          type="file"
          accept=".kc"
          id="file"
          onChange={this.handleFileUpload}
        />
        <label htmlFor="file">Choose a file</label>
        <span id="file_span">{this.state.selectedFile?.name}</span>
        <button id="proceed" onClick={this.importKeysFromFile}>
          {chrome.i18n.getMessage('popup_html_import')}
        </button>
      </div>
    );
  }
}

export default App;
