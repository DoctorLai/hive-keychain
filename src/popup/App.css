.App {
    height: 600px;
    width: 300px;
    font-family: Futura !important;
    font-weight: 400;
    background: url("/assets/images/hive-watermark.png") 0% 0% no-repeat
    padding-box, linear-gradient(330deg, #59626c 0%, #000000 100%) 0%
    0% no-repeat padding-box !important;

    color: white;

}