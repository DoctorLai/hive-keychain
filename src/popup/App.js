import React, { Component } from "react";

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>{chrome.i18n.getMessage("popup_html_new_password")}</h1>
      </div>
    );
  }
}

export default App;
