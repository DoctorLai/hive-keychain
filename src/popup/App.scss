@import './variable.scss';
@font-face {
  font-family: Futura;
  src: url('/assets/fonts/Futura-Boo.otf');
  font-weight: 400;
}

.App {
  height: 600px;
  width: 350px;
  font-family: Futura !important;
  font-weight: 400;
  background: url('/assets/images/hive-watermark.png') 0% 0% no-repeat
      padding-box,
    linear-gradient(330deg, #59626c 0%, #000000 100%) 0% 0% no-repeat
      padding-box !important;
  color: white;
  box-sizing: border-box;
  overflow: auto;

  &.homepage {
    background: url(/assets/images/hive-watermark.png) 0% 0% no-repeat
        padding-box,
      linear-gradient(to bottom, #000000, #59626c, #000000) 0% 0% no-repeat
        padding-box !important;
  }

  .page-container {
    display: grid;
    row-gap: 32px;
    grid-template-rows: 28px 1fr;
    height: 100%;
  }
  .divider {
    flex: 1;
  }
}

.react-dropdown-select-dropdown-handle {
  width: 48px;
  svg {
    width: 20px !important;
    height: 20px !important;
  }
}

.pretty.p-switch .state:before {
  top: calc((0% - (100% - 1em)) - 13%) !important;
}

.css-wmw4vi-ReactDropdownSelect {
  padding: 2px 0px !important;
}

.Toastify__progress-bar {
  bottom: unset;
  top: 0;
}

.Toastify__toast-icon {
  display: none !important;
}

.Toastify__progress-bar {
  background: white !important;
}

.Toastify__toast--error {
  background: $message-container-error !important;
}
.Toastify__toast--info {
  background: $message-container-info !important;
}
.Toastify__toast--success {
  background: $message-container-success !important;
}
