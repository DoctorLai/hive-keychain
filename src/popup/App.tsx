import {retrieveAccounts} from '@popup/actions/account.actions';
import {setMk} from '@popup/actions/mk.actions';
import {navigateTo} from '@popup/actions/navigation.actions';
import {RootState} from '@popup/store';
import React, {useEffect} from 'react';
import {connect, ConnectedProps} from 'react-redux';
import {BackgroundMessage} from 'src/background/background-message.interface';
import {BackgroundCommand} from 'src/reference-data/background-message-key.enum';
import {Screen} from 'src/reference-data/screen.enum';
import PopupUtils from 'src/utils/popup.utils';
import './App.css';
import {AddAccountRouterComponent} from './pages/add-account/add-account-router/add-account-router.component';
import {AppContainerComponent} from './pages/app-container/app-container.component';
import {ErrorMessageContainerComponent} from './pages/error-message-container/error-message-container.component';
import {SignInComponent} from './pages/sign-in/sign-in.component';
import {SignUpComponent} from './pages/sign-up/sign-up.component';

const App = ({
  setMk,
  mk,
  retrieveAccounts,
  accounts,
  navigateTo,
}: PropsFromRedux) => {
  useEffect(() => {
    PopupUtils.fixPopupOnMacOs();
  }, []);

  useEffect(() => {
    chrome.runtime.sendMessage({command: BackgroundCommand.GET_MK});
    chrome.runtime.onMessage.addListener(onSentBackMkListener);
  }, [setMk]);

  const onSentBackMkListener = (message: BackgroundMessage) => {
    if (message.command === BackgroundCommand.SEND_BACK_MK) {
      setMk(message.value);
      retrieveAccounts(message.value);
      chrome.runtime.onMessage.removeListener(onSentBackMkListener);
    }
  };

  const renderMainLayoutNav = () => {
    if (!mk) {
      if (accounts.length === 0) {
        return <SignUpComponent />;
      } else {
        return <SignInComponent />;
      }
    } else {
      if (accounts.length === 0) {
        navigateTo(Screen.ACCOUNT_PAGE_INIT_ACCOUNT);
        return <AddAccountRouterComponent />;
      } else {
        return <AppContainerComponent />;
      }
    }
  };

  return (
    <div className="App">
      {renderMainLayoutNav()}
      <ErrorMessageContainerComponent />
    </div>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    mk: state.mk,
    accounts: state.accounts,
  };
};

const connector = connect(mapStateToProps, {
  setMk,
  retrieveAccounts,
  navigateTo,
});
type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(App);
