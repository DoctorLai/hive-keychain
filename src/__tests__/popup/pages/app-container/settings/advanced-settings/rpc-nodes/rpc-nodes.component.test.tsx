import App from '@popup/App';
import { Icons } from '@popup/icons.enum';
import { Screen } from '@reference-data/screen.enum';
import '@testing-library/jest-dom';
import { act, cleanup, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import ariaLabelButton from 'src/__tests__/utils-for-testing/aria-labels/aria-label-button';
import arialabelCheckbox from 'src/__tests__/utils-for-testing/aria-labels/aria-label-checkbox';
import ariaLabelInput from 'src/__tests__/utils-for-testing/aria-labels/aria-label-input';
import ariaLabelSelect from 'src/__tests__/utils-for-testing/aria-labels/aria-label-select';
import initialStates from 'src/__tests__/utils-for-testing/data/initial-states';
import reactTestingLibrary from 'src/__tests__/utils-for-testing/rtl-render/rtl-render-functions';
import { HiveTxUtils } from 'src/utils/hive-tx.utils';
describe('rpc-nodes.component tests:\n', () => {
  afterEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
    cleanup();
  });
  describe('Switch rpc auto:\n', () => {
    beforeEach(async () => {
      await reactTestingLibrary.renderWithConfiguration(
        <App />,
        initialStates.iniStateAs.defaultExistent,
      );
      await act(async () => {
        await userEvent.click(screen.getByLabelText(ariaLabelButton.menu));
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.menuPreFix + Icons.SETTINGS),
        );
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.menuPreFix + Icons.RPC),
        );
      });
    });

    it('Must show rpc_nodes page', () => {
      expect(screen.getByLabelText(`${Screen.SETTINGS_RPC_NODES}-page`));
    });

    it('Must show add rpc button', async () => {
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(
            arialabelCheckbox.rpcNodes.select.automaticMode,
          ),
        );
      });
      expect(
        screen.getByLabelText(ariaLabelButton.rpcNodes.addRpc),
      ).toBeInTheDocument();
    });
  });
  describe('Not auto:\n', () => {
    beforeEach(async () => {
      await reactTestingLibrary.renderWithConfiguration(
        <App />,
        initialStates.iniStateAs.defaultExistent,
        {
          app: {
            localStorageRelated: {
              customData: {
                customSwitchAuto: false,
                customsRpcs: [
                  { uri: 'https://saturnoman.com/rpc', testnet: false },
                ],
              },
            },
          },
        },
      );
      await act(async () => {
        await userEvent.click(screen.getByLabelText(ariaLabelButton.menu));
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.menuPreFix + Icons.SETTINGS),
        );
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.menuPreFix + Icons.RPC),
        );
      });
    });

    it('Must hide add rpc button', async () => {
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(
            arialabelCheckbox.rpcNodes.select.automaticMode,
          ),
        );
      });
      expect(
        screen.queryByLabelText(ariaLabelButton.rpcNodes.addRpc),
      ).not.toBeInTheDocument();
    });

    it('Must show error if empty uri', async () => {
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.rpcNodes.addRpc),
        );
        await userEvent.type(
          screen.getByLabelText(ariaLabelInput.rpcNodes.uri),
          '{space}',
        );
        await userEvent.click(screen.getByLabelText(ariaLabelButton.save));
      });
      expect(
        await screen.findByText(
          chrome.i18n.getMessage('popup_html_rpc_missing_fields'),
          { exact: true },
        ),
      ).toBeInTheDocument();
    });

    it('Must show error if invalid uri', async () => {
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.rpcNodes.addRpc),
        );
        await userEvent.type(
          screen.getByLabelText(ariaLabelInput.rpcNodes.uri),
          'www.www.rpcNode',
        );
        await userEvent.click(screen.getByLabelText(ariaLabelButton.save));
      });
      expect(
        await screen.findByText(
          chrome.i18n.getMessage('html_popup_url_not_valid'),
          { exact: true },
        ),
      ).toBeInTheDocument();
    });

    it('Must show error if empty node chain Id', async () => {
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.rpcNodes.addRpc),
        );
        await userEvent.type(
          screen.getByLabelText(ariaLabelInput.rpcNodes.uri),
          'https://saturno.hive.com/rpc',
        );
        await userEvent.click(
          screen.getByLabelText(
            arialabelCheckbox.rpcNodes.select.addTesnetNode,
          ),
        );
        await userEvent.click(screen.getByLabelText(ariaLabelButton.save));
      });
      expect(
        await screen.findByText(
          chrome.i18n.getMessage('popup_html_rpc_missing_fields'),
          { exact: true },
        ),
      ).toBeInTheDocument();
    });

    it('Must add new rpc and show it in list', async () => {
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.rpcNodes.addRpc),
        );
        await userEvent.type(
          screen.getByLabelText(ariaLabelInput.rpcNodes.uri),
          'https://saturno.hive.com/rpc',
        );
        await userEvent.click(screen.getByLabelText(ariaLabelButton.save));
        await userEvent.click(
          screen.getByLabelText(ariaLabelSelect.rpcNode.selected),
        );
      });
      expect(
        await screen.findByText('https://saturno.hive.com/rpc', {
          exact: true,
        }),
      ).toBeInTheDocument();
    });

    it('Must add new rpc and set it as active rpc', async () => {
      const sSetRpc = jest.spyOn(HiveTxUtils, 'setRpc');
      await act(async () => {
        await userEvent.click(
          screen.getByLabelText(ariaLabelButton.rpcNodes.addRpc),
        );
        await userEvent.type(
          screen.getByLabelText(ariaLabelInput.rpcNodes.uri),
          'https://saturno.hive.com/rpc',
        );
        await userEvent.click(
          screen.getByLabelText(arialabelCheckbox.rpcNodes.select.setAsActive),
        );
        await userEvent.click(screen.getByLabelText(ariaLabelButton.save));
        await userEvent.click(
          screen.getByLabelText(ariaLabelSelect.rpcNode.selected),
        );
      });
      expect(
        screen.getByLabelText(
          ariaLabelSelect.rpcNode.selectItem.preFix +
            'https://saturno.hive.com/rpc',
        ),
      ).toBeInTheDocument();
      sSetRpc.mockRestore();
    });
  });
});
