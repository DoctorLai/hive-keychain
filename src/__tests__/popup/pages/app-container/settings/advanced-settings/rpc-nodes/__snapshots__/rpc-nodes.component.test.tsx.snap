// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rpc-nodes.component tests:
 Not auto:
 Must load component and match snapshot: rpc-nodes.component No Switch Auto 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          RPC Node
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          aria-label="rpc-nodes-page"
          class="rpc-nodes-page"
        >
          <p
            class="introduction"
          >
            You can choose which RPC node is used to read information and broadcast transactions from/to the Hive Blockchain.
          </p>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="false"
                  checked=""
                  id="checkbox-rpc-nodes-automatic-mode-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-rpc-nodes-automatic-mode-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-rpc-nodes-automatic-mode"
                class="label"
              >
                Switch automatically
              </div>
            </div>
            <div
              class="hint"
            >
              Will always select the best RPC node for you
            </div>
          </div>
          <div
            class="select-account-section"
          >
            <div>
              <div
                aria-expanded="true"
                aria-label="Dropdown select"
                class="react-dropdown-select select-rpc-node-select css-16vukcu e1gzf2xs0"
                color="#0074D9"
                direction="ltr"
                tabindex="0"
              >
                <div
                  class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
                >
                  <div
                    aria-label="selected-rpc-node-panel"
                    class="selected-rpc-node-panel"
                  >
                    <div
                      aria-label="selected-rpc-node"
                      class="selected-rpc-node"
                    >
                      DEFAULT
                    </div>
                  </div>
                </div>
                <div
                  class="react-dropdown-select-dropdown-handle css-4ft19d e1vudypg0"
                  color="#0074D9"
                  rotate="1"
                  tabindex="-1"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 40 40"
                  >
                    <path
                      d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                    />
                  </svg>
                </div>
                <div
                  aria-expanded="true"
                  class="react-dropdown-select-dropdown react-dropdown-select-dropdown-position-bottom css-1jemt82 e1qjn9k91"
                  role="list"
                  tabindex="-1"
                >
                  <div
                    aria-label="select-rpc-item-https://api.hive.blog/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://api.hive.blog/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://api.deathwing.me"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://api.deathwing.me 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://api.openhive.network/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://api.openhive.network/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://api.hivekings.com/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://api.hivekings.com/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://anyx.io/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://anyx.io/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://api.pharesim.me/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://api.pharesim.me/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://hived.emre.sh"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://hived.emre.sh 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://hived.hive-engine.com/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://hived.hive-engine.com/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://rpc.ausbit.dev"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://rpc.ausbit.dev 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://rpc.ecency.com"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://rpc.ecency.com 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://techcoderx.com"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://techcoderx.com 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://hive-api.arcange.eu/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://hive-api.arcange.eu/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://hived.privex.io/"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://hived.privex.io/ 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://hive.roelandp.nl"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://hive.roelandp.nl 
                    </div>
                  </div>
                  <div
                    aria-label="select-rpc-item-https://saturnoman.com/rpc"
                    class="select-rpc-item "
                  >
                    <div
                      class="rpc-name"
                    >
                      https://saturnoman.com/rpc 
                    </div>
                    <img
                      aria-label="button-erase-custom-rpc"
                      class="erase-button"
                      src="/assets/images/clear.png"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            aria-label="button-add-rpc"
            class="submit-button default "
          >
            <div
              class="button-label"
            >
              Add RPC 
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rpc-nodes.component tests:
 Switch rpc auto:
 Must load component and match snapshot: rpc-nodes.component Switch Auto 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          RPC Node
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          aria-label="rpc-nodes-page"
          class="rpc-nodes-page"
        >
          <p
            class="introduction"
          >
            You can choose which RPC node is used to read information and broadcast transactions from/to the Hive Blockchain.
          </p>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  checked=""
                  id="checkbox-rpc-nodes-automatic-mode-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-rpc-nodes-automatic-mode-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-rpc-nodes-automatic-mode"
                class="label"
              >
                Switch automatically
              </div>
            </div>
            <div
              class="hint"
            >
              Will always select the best RPC node for you
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
